driver-core/src/main/java/com/datastax/driver/core/CodecFactory.java

```diff
@@ -0,0 +1,250 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.reflect.TypeToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.datastax.driver.core.TypeCodec.*;
+
+import static com.datastax.driver.core.DataType.Name.*;
+
+/**
+ * A factory for {@link TypeCodec} instances.
+ * Instances of this interface are used by the {@link CodecRegistry} to create
+ * codecs on-the-fly.
+ */
+public interface CodecFactory {
```

Comment:
This is basically created out of the `maybeCreateCodec` methods in `CodecRegistry`.